hooks/hook.go

```diff
@@ -0,0 +1,37 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// The hooks package contains all hook runners supported by snapd.
+package hooks
+
+import (
+	"gopkg.in/tomb.v2"
+)
+
+// HookRef is a reference to a hook within a specific snap.
+type HookRef struct {
+	Snap string `json:"snap"`
```

Comment:
Without knowing all the hooks we intend on having, I can't think of a situation in which we'd want to run hooks on a non-active version of a snap. The only situation that may be problematic is if a snap is removed when a hook for it is queued up, but that's something that should be handled anyhow.

This probably needs more discussion: easy to add the revision, the question is if we need it or not.